logicalRequirements.md

```diff
@@ -182,6 +182,15 @@ __Example:__
 {"energyAtMost": 1}
 ```
 
+#### autoReserveTrigger object
+
+An `autoReserveTrigger` object represents a logical requirement for "auto" reserves to be triggered, which results in Samus' energy becoming equal to the amount of energy in reserves, and reserve energy becoming zero.
```

Comment:
- Added a note now to clarify that the energy gets capped to capacity.
- Added properties `minReserveEnergy` and `maxReserveEnergy` to `autoReserveTrigger`. I'm not sure if this completely addresses your comment, but it can hopefully help. For example, `minReserveEnergy` could be used to ensure that enough energy is in reserves to make sure that i-frames run out (e.g. to get a second damage boost), while `maxReserveEnergy` could be used to make sure that i-frames do not run out before getting clear of an enemy, to avoid a double hit.

Reply:
Looking at these Rinka rooms:
The crystal flash fills health and reserves, then you lower your health by taking rinka damage repeatedly.
The `autoReserveTrigger` should be followed by a second damage instance which is the hit from IFrames falling off before the refill finishes.  Then you start xray climbing and take a rinka hit while climbing.

The minReserveEnergy example helps if the next hit is going to do less than 60 damage, and wasn't thinking about the situation, but it does come up.
The maxReserveEnergy case overlaps with using pause abuse to mostly ignore chunky hits.

Reply:
The Rinkas are unusual in that they continue moving during the auto reserve, so they end up being out of the way by the time that finishes, meaning you don't get double hit.

Reply:
> * A damage source could be added as a parameter for determining double hits.  Something like 60energy refilled (minus buffer) for enemy contact damage, and 40ish for spikes.  Planned double hits almost fit in the `requires` already, a strat using those would force the second hit.  So it's the general damage taken as part of strats that cause energy to reach 0 except that would be a separate kind of passive autoReserveTrigger.

Changing `autoReserveTrigger` to cover more situations can be done with a separate PR.